samples/e2e-mnist/e2e-mnist.py

```diff
@@ -0,0 +1,315 @@
+      {"name": "--tf-batch-size", "parameterType": "discrete", "feasibleSpace": {"list": ["16", "32", "64"]}},
+    ]
+    rawTemplate = {
+      "apiVersion": "kubeflow.org/v1",
+      "kind": "TFJob",
+      "metadata": {
+         "name": "{{.Trial}}",
+         "namespace": "{{.NameSpace}}"
+      },
+      "spec": {
+        "tfReplicaSpecs": {
+          "Chief": {
+            "replicas": 1,
+            "restartPolicy": "OnFailure",
+            "template": {
+              "spec": {
+                "containers": [
+                {
+                  "command": [
+                    "sh",
+                    "-c"
+                  ],
+                  "args": [
+                    "python /opt/model.py --tf-train-steps=2000 {{- with .HyperParameters}} {{- range .}} {{.Name}}={{.Value}} {{- end}} {{- end}}"
+                  ],
+                  "image": "liuhougangxa/tf-estimator-mnist",
+                  "name": "tensorflow"
+                }
+                ]
+              }
+            }
+          },
+          "Worker": {
+            "replicas": 3,
+            "restartPolicy": "OnFailure",
+            "template": {
+              "spec": {
+                "containers": [
+                {
+                  "command": [
+                    "sh",
+                    "-c"
+                  ],
+                  "args": [ 
+                    "python /opt/model.py --tf-train-steps=2000 {{- with .HyperParameters}} {{- range .}} {{.Name}}={{.Value}} {{- end}} {{- end}}"
+                  ],
+                  "image": "liuhougangxa/tf-estimator-mnist",
+                  "name": "tensorflow"
+                }
+                ]
+              }
+            }
+          }
+        }
+      }
+    }
+    
+    trialTemplate = {
+      "goTemplate": {
+        "rawTemplate": json.dumps(rawTemplate)
+      }
+    }
+
+    metricsCollectorSpec = {
+      "source": {
+        "fileSystemPath": {
+          "path": "/tmp/tf",
+          "kind": "Directory"
+        }
+      },
+      "collector": {
+        "kind": "TensorFlowEvent"
+      }
+    }
+
+    katib_experiment_launcher_op = components.load_component_from_url('https://raw.githubusercontent.com/kubeflow/pipelines/master/components/kubeflow/katib-launcher/component.yaml')
+    op1 = katib_experiment_launcher_op(
+            experiment_name=name,
+            experiment_namespace=namespace,
+            parallel_trial_count=3,
+            max_trial_count=12,
+            objective=str(objectiveConfig),
+            algorithm=str(algorithmConfig),
+            trial_template=str(trialTemplate),
+            parameters=str(parameters),
+            metrics_collector=str(metricsCollectorSpec),
+            # experiment_timeout_minutes=experimentTimeoutMinutes,
+            delete_finished_experiment=False)
+
+    # step2: create a TFJob to train your model with best hyperparameter tuned by Katib
+    tfjobjson_template = Template("""
```

Comment:
I agree. This whole example wasn't designed with good python coding practice. Since the original author is no longer maintaining it. I can rewrite this whole example with a better design to do the same job.

Reply:
FYI many of the samples in KFP are not written by python programmers. So for the examples we pull from KFP, often times you will see bad practices but it just works when compile.

Reply:
For the json format check, it's currently done at here
https://github.com/kubeflow/kfp-tekton/pull/316/files#diff-dd6cd42fc3a3a005983202abafea90fda312af2d8b52d94dc5df7c1b8f34b4caR258

I agree we should make this example to look better, but i will need to do some refactoring so I will put it in my TODO list.